src/osd/PrimaryLogPG.cc

```diff
@@ -6927,6 +6927,9 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	    oi.manifest.chunk_map[src_offset].flags |=
 	      chunk_info_t::FLAG_HAS_REFERENCE;
 	  }
+	  if (need_reference && pool.info.fingerprint != pg_pool_t::TYPE_FINGERPRINT_NONE) {
+	    oi.manifest.chunk_map[src_offset].flags |= chunk_info_t::FLAG_HAS_FINGERPRINT;
+	  }
```

Comment:
What does this chunk_info_t flag actually tell us? Just that the chunk target happens to be named based on a hash of the content? So far we don't ever look at it. Can you think of a case where we will need it later?

Reply:
@liewegas Yes, as you know, this is post-dedup. So, at least, we should know which object need to be deduped when flushing starts. I think set-chunk interface we already have can do it easily because it has reference and fixed chunk property. Without this FLAG_HAS_FINGERPRINT flag, how can we distinguish between normal chunk and dedup chunk ?